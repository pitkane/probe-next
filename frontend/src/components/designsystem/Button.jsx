/* eslint-disable react/button-has-type */

import React from "react";
import PropTypes from "prop-types";

const Button = props => {
  const { type, value, primary, secondary, danger, onClick } = props;

  const className = ["uk-button", "uk-button-large", "uk-width-1-1"];

  if (primary) {
    className.push("uk-button-primary");
  }
  if (secondary) {
    className.push("uk-button-secondary");
  }
  if (danger) {
    className.push("uk-button-danger");
  }

  return (
    <div className="uk-margin">
      <button type={type} className={className.join(" ")} onClick={onClick}>
        {value}
      </button>
    </div>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  value: PropTypes.string.isRequired,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  danger: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  type: "button",
  primary: false,
  secondary: false,
  danger: false
};

export default Button;
