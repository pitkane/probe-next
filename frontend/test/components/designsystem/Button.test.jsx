import React from "react";
import renderer from "react-test-renderer";

import Button from "../../../src/components/designsystem/Button";

describe("Button", () => {
  it("renders correctly", () => {
    const component = <Button onClick={() => {}} value="lol" />;
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
