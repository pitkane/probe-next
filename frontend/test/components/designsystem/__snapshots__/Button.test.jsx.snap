// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="uk-margin"
>
  <button
    className="uk-button uk-button-large uk-width-1-1"
    onClick={[Function]}
    type="button"
  >
    lol
  </button>
</div>
`;
